src/settings/eeprom/rf_channel.rs

```diff
@@ -113,40 +144,44 @@ impl Default for ChannelSettings {
             //
             // All of the power meters are preceded by attenuators which are incorporated in
             // the offset.
-            output_power_transform: LinearTransformation::new(1.0 / 0.035, -35.6 + 19.8 + 10.0),
+            output_power_transform: Leaf(LinearTransformation::new(
+                1.0 / 0.035,
+                -35.6 + 19.8 + 10.0,
+            )),
 
             // The input power and reflected power detectors have an op-amp gain of 1.5
-            reflected_power_transform: LinearTransformation::new(
+            reflected_power_transform: Leaf(LinearTransformation::new(
                 1.0 / 1.5 / 0.035,
                 -35.6 + 19.8 + 10.0,
-            ),
+            )),
 
-            input_power_transform: LinearTransformation::new(1.0 / 1.5 / 0.035, -35.6 + 8.9),
+            input_power_transform: Leaf(LinearTransformation::new(1.0 / 1.5 / 0.035, -35.6 + 8.9)),
         }
     }
 }
 
 impl ChannelSettings {
-    fn validate_bias_voltage(&mut self, new: f32) -> Result<f32, &'static str> {
-        if (0.0..=platform::BIAS_DAC_VCC).contains(&(-1.0 * new)) {
-            Ok(new)
-        } else {
-            Err("Bias voltage out of range")
-        }
+    fn validate_bias_voltage(&mut self, depth: usize) -> Result<usize, &'static str> {
+        *self.bias_voltage = -(-*self.bias_voltage).clamp(0.0, platform::BIAS_DAC_VCC);
+        Ok(depth)
     }
 
-    fn validate_output_interlock(&mut self, new: f32) -> Result<f32, &'static str> {
+    fn validate_output_interlock_threshold(&mut self, depth: usize) -> Result<usize, &'static str> {
         // Verify the output interlock is within acceptable values.
-        if new >= platform::MAX_OUTPUT_POWER_DBM {
-            return Err("Output interlock threshold too high");
-        }
+        *self.output_interlock_threshold = self
+            .output_interlock_threshold
+            .clamp(0.0, platform::MAX_OUTPUT_POWER_DBM);
```

Comment:
This now allows `47.0` to be set, whereas before this would fail due to the `>=`.